packages/core/__tests__/interpreter/cancellation.js

```diff
@@ -738,3 +738,29 @@ test('cancel should support for self cancellation', () => {
     expect(actual).toEqual(expected)
   })
 })
+
+test('should bubble an exception thrown during cancellation', () => {
+  const middleware = sagaMiddleware()
+  createStore(() => ({}), {}, applyMiddleware(middleware))
+
+  const expectedError = new Error('child error')
+  function* child() {
+    try {
+      yield io.delay(1000);
+    } finally {
+      // eslint-disable-next-line no-unsafe-finally
+      throw expectedError;
+    }
+  }
+  function* worker() {
+    const taskA = yield io.fork(child);
+    yield io.delay(100)
+    yield io.cancel(taskA);
+  }
+
+  return middleware.run(worker)
+    .toPromise()
+    .catch((error) => {
+      expect(error).toEqual(expectedError)
```

Comment:
Actually this promise will resolves to `undefined`, so the catch-callback does not run and the test passes.

If you change this to ...
```javascript
const promise = middleware.run(worker).toPromise()
return expect(promise).rejects.toBe(expectedError)
```
... then the test will fail.